google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1007,7 +1017,11 @@ void populateCondition(Rule.Condition condition) {
 
     @Override
     void populateCondition(Rule.Condition condition) {
-      throw new UnsupportedOperationException();
+      log.warning(
```

Comment:
This change doesn't appear to be exercised by the updated test. (Ran the test with code coverage and it didn't hit this line.

Reply:
this change is for the old, deprecated version of lifecycle rules, but i can add a test for that if you think it's necessary